orchagent/portsorch.cpp

```diff
@@ -4040,6 +4136,139 @@ void PortsOrch::doPortTask(Consumer &consumer)
                         );
                     }
                 }
+
+                if (pCfg.pt_intf_id.is_set)
+                {
+                    if (p.m_pt_intf_id != pCfg.pt_intf_id.value)
+                    {
+                        /*
+                         * First, let's check the Path Tracing Interface ID configured for the port.
+                         *
+                         * Path Tracing Interface ID > 0 -> Path Tracing ENABLED on the port
+                         * Path Tracing Interface ID == 0 -> Path Tracing DISABLED on the port
+                         */
+                        if (pCfg.pt_intf_id.value != 0)
+                        {
+                            /* Path Tracing ENABLED case */
+
+                            /*
+                             * First, let's check if a TAM object is already assigned to the port.
+                             */
+                            if (m_portPtTam.find(p.m_alias) == m_portPtTam.end())
+                            {
+                                /*
+                                 * The port does not have a TAM object assigned to it.
+                                 *
+                                 * Let's create a new TAM object (if we don't already have one)
+                                 * and assign it to the port.
+                                 */
+                                if (m_ptTam == SAI_NULL_OBJECT_ID)
+                                {
+                                    if (!createPtTam())
+                                    {
+                                        SWSS_LOG_ERROR(
+                                            "Failed to create TAM object for Path Tracing"
+                                        );
+                                        it++;
+                                        continue;
+                                    }
+                                }
+
+                                if (!setPortPtTam(p, m_ptTam))
```

Comment:
We could move TAM create, setting TAM to port and refcount update to a different function.

Reply:
@kperumalbfn I moved all the logic related to the TAM to separate functions: https://github.com/sonic-net/sonic-swss/pull/2903/commits/fc6b5ba78827605012b97dac788eb44f599777f6